src/scripts/main.js

```diff
@@ -1,3 +1,275 @@
+const selectOffice = document.createElement('select');
+
+selectLabel.setAttribute('data-qa', 'office');
+selectLabel.innerText = 'Office:';
+form.append(selectLabel);
+
+selectOffice.setAttribute('data-qa', 'office');
+selectOffice.setAttribute('name', 'office');
+selectLabel.append(selectOffice);
+
+cities.forEach((citi) => {
+  const optionCiti = document.createElement('option');
+
+  optionCiti.setAttribute('value', citi);
+  optionCiti.innerText = citi;
+  selectOffice.append(optionCiti);
+});
+
+// age input
+const ageLabel = document.createElement('label');
+const ageInput = document.createElement('input');
+
+ageLabel.setAttribute('data-qa', 'age');
+ageLabel.innerText = 'Age:';
+form.append(ageLabel);
+
+ageInput.setAttribute('name', 'age');
+ageInput.setAttribute('type', 'number');
+ageLabel.append(ageInput);
+
+// salary input
+const salaryLabel = document.createElement('label');
+const salaryInput = document.createElement('input');
+
+salaryLabel.setAttribute('data-qa', 'salary');
+salaryLabel.innerText = 'Salary:';
+form.append(salaryLabel);
+
+salaryInput.setAttribute('name', 'salary');
+salaryInput.setAttribute('type', 'number');
+salaryLabel.append(salaryInput);
+
+// button form
+const button = document.createElement('button');
+
+button.setAttribute('name', 'saveToTable');
+button.setAttribute('type', 'submit');
+button.innerText = 'Save to table';
+form.append(button);
+
+// button submit
+const formData = document.querySelector('.new-employee-form');
+
+formData.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+
+  const fData = {
+    name: data.get('name'),
+    position: data.get('position'),
+    office: data.get('office'),
+    age: data.get('age'),
+    salary: new Intl.NumberFormat('en-US', {
+      style: 'currency',
+      currency: 'USD',
+      minimumFractionDigits: 0,
+      maximumFractionDigits: 0,
+    }).format(data.get('salary')),
+  };
+
+  // validator
+  if (!validName(fData.name)) {
+    pushNotification(
+      10,
+      10,
+      'Wrong Name',
+      'Error.\n ' +
+        'The name must consist of letters and be longer than 4 characters',
+      'error',
+    );
+  } else if (!validAge(fData.age)) {
+    pushNotification(
+      10,
+      10,
+      'Wrong Age',
+      'Error.\n ' + 'Аge must be between 18 and 90.',
+      'error',
+    );
+  } else {
+    const tr = document.createElement('tr');
+
+    [...thead.querySelectorAll('th')].forEach((header) => {
+      const key = header.innerText.toLocaleLowerCase();
+      const td = document.createElement('td');
+
+      td.innerText = fData[key];
+      tr.append(td);
+    });
+
+    tbody.append(tr);
+    form.reset();
+
+    pushNotification(
+      10,
+      10,
+      'Successful addition',
+      'Success.\n ' + 'Record successfully added to the table.',
+      'success',
+    );
+  }
+});
+
+const validName = (dataName) => {
+  const namePattern = /^[a-zA-Z]{4,}$/;
```

Comment:
The `validName` function uses a regex pattern that only allows letters and requires at least 4 characters. This might be too restrictive if names can include spaces or special characters. Consider updating the regex pattern to accommodate such cases if needed.